corehq/apps/api/resources/v0_5.py

```diff
@@ -1344,7 +1344,8 @@ def get_datasource_data(request, config_id, domain):
     config, _ = get_datasource_config(config_id, domain)
     datasource_adapter = get_indicator_adapter(config, load_source='export_data_source')
     request_params = get_request_params(request).params
-    if "limit" not in request_params or int(request_params["limit"]) > EXPORT_DATASOURCE_DEFAULT_PAGINATION_LIMIT:
+    request_params["limit"] = request.GET.dict().get("limit", EXPORT_DATASOURCE_DEFAULT_PAGINATION_LIMIT)
```

Comment:
The benefit from the reverted change was to return all params in `request_params`, so I'm just adding the `limit` param here (if it's not already there) so we can still enjoy a similar convenience. Could have just done this from the start I guess